File: zerver_backend.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# Sample data
items = [
    {"id": 1, "name": "item1", "price": 10.99},
    {"id": 2, "name": "item2", "price": 12.99},
]

# Get all items
@app.route('/items', methods=['GET'])
def get_items():
    return jsonify(items)

# Get a specific item by id
@app.route('/items/<int:item_id>', methods=['GET'])
def get_item(item_id):
    item = next((item for item in items if item['id'] == item_id), None)
    return jsonify(item) if item else ('', 404)

# Create a new item
@app.route('/items', methods=['POST'])
def create_item():
    new_item = request.get_json()
    new_item['id'] = max(item['id'] for item in items) + 1
    items.append(new_item)
    return jsonify(new_item), 201

# Update an existing item
@app.route('/items/<int:item_id>', methods=['PUT'])
def update_item(item_id):
    item = next((item for item in items if item['id'] == item_id), None)
    if item:
        data = request.get_json()
        item.update(data)
        return jsonify(item)
    return ('', 404)

# Delete an item
@app.route('/items/<int:item_id>', methods=['DELETE'])
def delete_item(item_id):
    global items
    items = [item for item in items if item['id'] != item_id]
    return ('', 204)

if __name__ == '__main__':
    app.run(host='10.101.145.133',debug=True)

File: zerver_backend.js
const express = require('express');
const app = express();
const port = 3000;

app.use(express.json());

// Sample data
let items = [
    { id: 1, name: 'item1', price: 10.99 },
    { id: 2, name: 'item2', price: 12.99 },
];

// Get all items
app.get('/items', (req, res) => {
    res.json(items);
});

// Get a specific item by id
app.get('/items/:id', (req, res) => {
    const item = items.find(i => i.id === parseInt(req.params.id));
    if (item) {
        res.json(item);
    } else {
        res.status(404).send('Item not found');
    }
});

// Create a new item
app.post('/items', (req, res) => {
    const newItem = req.body;
    newItem.id = items.length ? Math.max(...items.map(i => i.id)) + 1 : 1;
    items.push(newItem);
    res.status(201).json(newItem);
});

// Update an existing item
app.put('/items/:id', (req, res) => {
    const item = items.find(i => i.id === parseInt(req.params.id));
    if (item) {
        Object.assign(item, req.body);
        res.json(item);
    } else {
        res.status(404).send('Item not found');
    }
});

// Delete an item
app.delete('/items/:id', (req, res) => {
    const itemIndex = items.findIndex(i => i.id === parseInt(req.params.id));
    if (itemIndex !== -1) {
        items.splice(itemIndex, 1);
        res.status(204).send();
    } else {
        res.status(404).send('Item not found');
    }
});

app.listen(port, '10.101.145.133' () => {
    console.log(`Server running at http://10.101.145.133:${port}/`);
});